src/native/mqtt_connection.c

```diff
@@ -538,14 +568,18 @@ static void s_deliver_ack_success(struct mqtt_jni_async_callback *callback) {
     AWS_FATAL_ASSERT(callback->connection);
 
     if (callback->async_callback) {
-        JNIEnv *env = aws_jni_get_thread_env(callback->connection->jvm);
+        /********** JNI ENV ACQUIRE **********/
+        JNIEnv *env = aws_jni_acquire_thread_env(callback->connection->jvm);
```

Comment:
`s_deliver_ack_success()` is a helper function, not a callback, we could just pass env in as an argument